dspace-api/src/main/java/org/dspace/importer/external/arxiv/metadatamapping/contributor/ArXivIdMetadataContributor.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.arxiv.metadatamapping.contributor;
+
+import java.util.Collection;
+
+import org.apache.axiom.om.OMElement;
+import org.dspace.importer.external.metadatamapping.MetadatumDTO;
+import org.dspace.importer.external.metadatamapping.contributor.MetadataContributor;
+import org.dspace.importer.external.metadatamapping.contributor.SimpleXpathMetadatumContributor;
+
+/**
+ * Arxiv specific implementation of {@link MetadataContributor}
+ * Responsible for generating the ArXiv Id from the retrieved item.
+ * 
+ * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
+ *
+ */
+public class ArXivIdMetadataContributor extends SimpleXpathMetadatumContributor {
+
+    /**
+     * Retrieve the metadata associated with the given object.
+     * Depending on the retrieved node (using the query), different types of values will be added to the MetadatumDTO
+     * list
+     *
+     * @param t A class to retrieve metadata from.
+     * @return a collection of import records. Only the identifier of the found records may be put in the record.
+     */
+    @Override
+    public Collection<MetadatumDTO> contributeMetadata(OMElement t) {
+        Collection<MetadatumDTO> values = super.contributeMetadata(t);
+        parseValue(values);
+        return values;
+    }
+
+    /**
+     * ArXiv doesn't return the item id. We have to get this from the path parameter
```

Comment:
It will be called, as the same as contributeMetadata, but "values" will be an empty array in any cases except on this line of the ArXiv XML (e.g.):
```
    <id>http://arxiv.org/abs/1911.11405v1</id>
```
because the OMElement have to match the XPath expression "ns:id" as defined in arxiv-integration.xml on dc.identifier.other metadatum.

In the matching case, values.value is http://arxiv.org/abs/1911.11405v1, where `1911.11405v1`. parseValue manipulate this string in order to extract the id and save it as metadatum value.

Reply:
@pasqualecvl : Please add these details (or a summary of them) into the JavaDocs themselves, as it is a much better description of what the method is doing.  Even simply adding a line that says something like: `ArXiv returns a full URL as the "id", e.g.  http://arxiv.org/abs/1911.11405v1. This method parses out the identifier from the end of the URL, e.g. 1911.11405v1`